sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/DialogStateResult.java

```diff
@@ -87,4 +87,38 @@ public String getDialogId() {
     public String getOperationContext() {
         return operationContext;
     }
+
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
+        jsonWriter.writeStartObject();
+        jsonWriter.writeStringField("dialogId", this.dialogId);
```

Comment:
Yes, it's intentional, the context is: This model always instantiated from the auto generated type `DialogStateResponse` (see Ctr) and never populated or exposed the other two fields (dialogInputType, dialogOptions). Also, given this is a fluent model type (wrapping the generated wire type ` DialogStateResponse`), I don't think it ever directly used to deserialize from a wire response, even if someone do so, regardless of azure-json or Jackson, the `dialogInputType` & `dialogOptions` won’t be populated because there are no such properties in wire response.

Btw, I don’t know why this fluent model is ever be a serde to begin with, but don’t want to break any.

Reply:
Mentioned this observation [here](https://github.com/Azure/azure-sdk-for-java/issues/40903) - Communication SDK owners to evaluate the need for serde enabling custom models.